### ISSUE TITLE
Provide rainbow-delimiters in Gnu Elpa

### ISSUE THREAD

**Issue by ghost** [jlr/rainbow-delimiters#17]
Hi,

I'm a happy user of your superb rainbow-delimiters.el.

Some time ago there was a discussion about it in gnu.emacs.bugs, under
bug#13156.  Stefan Monnier mentioned that he would be happy if
rainbow-delimiters.el would be available in Gnu Elpa (the official Emacs
Lisp Package Archive):

  http://elpa.gnu.org/

Do you agree that it should be provided there?  If you do, you (and your
employer if appropriate) would need to sign the copyright papers for
that purpose, because Gnu Elpa packages are considered as part of the
Emacs distribution.

I am able to help you with the process, although I'm not really an
insider.  But it was not hard in my case and didn't took long.

Thanks!

Michael.

**Comment by jlr** [comment]
Yes, I would very much like to see rainbow-delimiters in Elpa. I'll have my desktop computer down where I've moved within a month and would like to move forward with integrating the various bug fixes and feature requests that have piled up since I've been away. Then I'd like to get it in Elpa.

The mode was written with Clojure in mind, so works perfectly with its escape chars to my knowledge. But other language modes need to be added. We can integrate rainbow-delimiters in Elpa once that's set up but it would be nice to have it working with most languages if it's going to be in Elpa.

In either case, I will happily deal with the paperwork involved with this. If you want to see it integrated sooner, I can deal with it in the very near future, or we can wait maybe 1.5months and I'll have done some new work on the mode; we could integrate it then. Either way is OK with me.

**Comment by ghost** [comment]
If you are sure that you can deal with it in the next months, and that the library is stable enough after your additional work, we can wait, of course.  But I can't withstand to say that I (personally) like to have it in Gnu Elpa as soon as possible.  Your more recent version will be available via Melpa anyway (did you know that?).

**Comment by purcell** [comment]
No need to wait before putting it in ELPA once the paperwork is done -- `rainbow-delimiters.el` is perfectly usable in its current form.

But I do wonder if there's much practical value to it, unless you view ELPA as a route to inclusion in Emacs itself. I doubt there are many people who have a recent Emacs and are using `package.el` to install packages, but haven't also enabled either Marmalade or MELPA.

**Comment by ghost** [comment]
Of course, it makes no difference to the user.  But Gnu Elpa is considered being a part of Emacs, and rainbow-delimiters is so elementary that it should not be just an addon, but a part of the thing.

The original issue in bug#13156 I cited was that a user requested a special face for parantheses.  rainbow-delimiters offers a superset of that, and when it's added to Emacs, the dev people don't need to implement a second solution for making parens more readable themselves.  That's all.

**Comment by Fanael** [comment]
> you would need to sign the copyright papers for that purpose

Which is something I won't do. It's too much trouble for zero gain. Sorry.

**Comment by ghost** [comment]
Ok, thanks.  It's a pity, really.
